csi/controller_server.go

```diff
@@ -0,0 +1,269 @@
+package csi
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	csi "github.com/container-storage-interface/spec/lib/go/csi/v0"
+	csicommon "github.com/kubernetes-csi/drivers/pkg/csi-common"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	longhornclient "github.com/rancher/longhorn-manager/client"
+	"github.com/rancher/longhorn-manager/util"
+)
+
+const (
+	oneGB               = 1073741824
+	timeoutAttachDetach = 30 * time.Second
```

Comment:
On a heavily loaded machine, attaching may take longer. Make it 120s by default for now.

Reply:
Done